test/unit-tests/components/views/rooms/memberlist/MemberListView-test.tsx

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2024 New Vector Ltd.
+Copyright 2022 The Matrix.org Foundation C.I.C.
+Copyright 2021 Šimon Brandner <simon.bra.ag@gmail.com>
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
+Please see LICENSE files in the repository root for full details.
+*/
+
+import React, { act } from "react";
+import { waitFor } from "jest-matrix-react";
+import { Room, RoomMember, MatrixEvent } from "matrix-js-sdk/src/matrix";
+
+import { filterConsole } from "../../../../../test-utils";
+import { Rendered, renderMemberList } from "./common";
+
+jest.mock("../../../../../../src/customisations/helpers/UIComponents", () => ({
+    shouldShowComponent: jest.fn(),
+}));
+
+type Children = (args: { height: number; width: number }) => React.JSX.Element;
+jest.mock("react-virtualized", () => {
+    const ReactVirtualized = jest.requireActual("react-virtualized");
+    return {
+        ...ReactVirtualized,
+        AutoSizer: ({ children }: { children: Children }) => children({ height: 1000, width: 1000 }),
+    };
+});
+jest.spyOn(HTMLElement.prototype, "offsetHeight", "get").mockReturnValue(1500);
+jest.spyOn(HTMLElement.prototype, "offsetWidth", "get").mockReturnValue(1500);
+
+describe("MemberListView and MemberlistHeaderView", () => {
+    filterConsole(
+        "Age for event was not available, using `now - origin_server_ts` as a fallback. If the device clock is not correct issues might occur.",
+    );
+
+    function memberString(member: RoomMember): string {
+        if (!member) {
+            return "(null)";
+        } else {
+            const u = member.user;
+            return (
+                "(" +
+                member.name +
+                ", " +
+                member.powerLevel +
+                ", " +
+                (u ? u.lastActiveAgo : "<null>") +
+                ", " +
+                (u ? u.getLastActiveTs() : "<null>") +
+                ", " +
+                (u ? u.currentlyActive : "<null>") +
+                ", " +
+                (u ? u.presence : "<null>") +
+                ")"
+            );
+        }
+    }
+
+    function expectOrderedByPresenceAndPowerLevel(
+        memberListRoom: Room,
+        memberTiles: NodeListOf<Element>,
+        isPresenceEnabled: boolean,
+    ) {
+        let prevMember: RoomMember | undefined;
+        for (const tile of memberTiles) {
+            const memberA = prevMember;
+            const memberB = memberListRoom.currentState.members[tile.getAttribute("aria-label")!.split(" ")[0]];
+            prevMember = memberB; // just in case an expect fails, set this early
+            if (!memberA) {
+                continue;
+            }
+
+            console.log("COMPARING A VS B:", memberString(memberA), memberString(memberB));
```

Comment:
is this still wanted?

Reply:
Yep, will be useful if we ever need to debug this test.